### PR TITLE
Update tfrecord_reader.py

### PR DESCRIPTION
**Pull request by mohazahran**
Uncommenting this line to skip bad queries without throwing an error.
.apply(data.experimental.ignore_errors())

### CODE REVIEW

File: python/ml4ir/base/data/tfrecord_reader.py [hunk 1]
```diff
@@ -797,7 +797,7 @@ def read(
         # Parallel calls set to AUTOTUNE: improved training performance by 40% with a classification model
         dataset = (
             dataset.map(parse_fn, num_parallel_calls=tf.data.experimental.AUTOTUNE)
-            # .apply(data.experimental.ignore_errors())
+             .apply(data.experimental.ignore_errors())
```
Comment (balikasg) [inline python/ml4ir/base/data/tfrecord_reader.py:800]: Can you share some context on why this was commented and why we are now uncommenting? 

LGTM though. 

A separate issue: 
- On the creation of the dataset here, are we applying any randomization? In other words does the model see the data always in the same order or not? I am looking at this [for example](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#shuffle)

**Review by balikasg — APPROVED** [review]
LGTM; please consider adding some description on the PR to better stream-line this. 
Also, I am wondering on shuffling, what do you think?

File: python/ml4ir/base/data/tfrecord_reader.py (hunk 1 above)
Comment (mohazahran) [reply python/ml4ir/base/data/tfrecord_reader.py:800]: I'm uncommenting this line to skip bad queries without throwing an error.

Offline, I believe the model see the data in the same order. However, running on Flowsnake, I believe there is some "un-planned" shuffling that happens perhaps due to spark.

Comment (mohazahran) [reply python/ml4ir/base/data/tfrecord_reader.py:800]: FWIW, I think shuffling the data per epoch would produce a better model.

Comment (balikasg) [reply python/ml4ir/base/data/tfrecord_reader.py:800]: Do you want to add it?

Comment (arvindsrikantan) [reply python/ml4ir/base/data/tfrecord_reader.py:800]: +1 on shuffling

**Pull request merged**